support/xspec-dev/testing.md

```diff
@@ -0,0 +1,24 @@
+# Testing
+
+This work in support of XSpec is in need of more rigorous testing.
+
+Interactive testing has been performed with all scripts.
+
+But requirements and specifications are loose, having been driven up to this point by functional requirements faced by the metaschema-xslt project - for a flexible, performant runtime that works independently of licensed software, with developer-friendly controls from the command line and inside CI/CD. Specifically, this means a number of issues, including testing, that bear on full integration back into XSpec have been set aside for the present.
+
+Some experience in application will also be helpful in prioritizing the testing work to be done.
+
+Meanwhile, some interesting questions, both practical and theoretical, are posed by the need to test these capabilities, especially outside particular deployments.
+
+For which capabilities can we write XSpec or meaningfully test and confirm (expose and document) in other ways, including
+
+  - `xsl:message` behavior including terminate="yes"
+  - side effects such as `xsl:result-document`?
```

Comment:
You probably already know that [external transformations](https://github.com/xspec/xspec/wiki/External-Transformation) can handle XSLT that includes `xsl:result-document`. Beyond just not erroring out, I guess a test scenario can check availability of, or read the contents of, the file you expect to have been created.

Reply:
I did not know that but it makes sense, thanks @galtm, very helpful indeed.